### ISSUE TITLE
How long it takes to Sync when running a node on a new computer??

### ISSUE THREAD

**Issue by ariesunny** [nanocurrency/raiblocks#833]
**Description of bug:**

Hi, I have been running a NANO node with version 12.1 for a week but it seems still not sync finished, the wallet shows the block number like 719499(7479560), So I am wondering how long it may take to sync when  running a node on a new computer?

![image](https://user-images.githubusercontent.com/30827440/39360352-8220c388-4a50-11e8-97eb-8ba2c9e6e469.png)

I checked part of the nano code, as I can see the two number shows on the wallet like the pic above, the first number is the Blocks  which has been verified and the second one is the Blocks which has been pulled from other nodes but not verified yet  --- saved the unchecked_cache in the code. And it seems all the Blocks in the unchecked_cache need to verified again and then saved, each block verify need to broadcast 4 times with a period of 16 seconds, so roughly it does will take a long time to verify all the blocks and will also create a lot network broadcast pressure, is it really necessary to run a full procedure to verify all the blocks again on every new launch node ?

Please correct me if the understanding is wrong, thank you so much!

**Steps to reproduce the issue:**
1. Run a fresh new node with on a new computer with no old NANO data on it.

**Describe the results you received:**
A week passed, but still on sync.

**Describe the results you expected:**
I am wondering how long to sync when  running a node on a new computer?

**Additional information you deem important (e.g. issue happens only occasionally):**

**Environment**:
Windows

**logs**

**Comment by rotilho** [comment]
Connected with 823

Glad to hear some explanation about underlying implementation

**Comment by PlasmaPower** [comment]
Blocks pulled from bootstrapping are not considered active. I believe the primary issue is DB and disk performance.

**Comment by PlasmaPower** [comment]
Blocks pulled from bootstrapping are not considered active. I believe the primary issue is DB and disk performance.

**Comment by ariesunny** [comment]
@PlasmaPower  Thank you for reply, I noticed that there is a modification in V12 that to check if the block is recently arrived, if it is not ,it will not  add to election list, Is this what you said "Blocks pulled from bootstrapping are not considered active." ?  BTW, it seems the Sync on my PC works very slowly and nearly no progress in the last two days, My PC with i5 CPU and SSD disk, Any idea what is go on?

**Comment by PlasmaPower** [comment]
It's always been that way. We have just changed around the exact mechanics of it recently

**Comment by PlasmaPower** [comment]
You could try restarting the wallet. I'm not sure what's going wrong in your specific case. Logs may help.

**Comment by ariesunny** [comment]
![_20180427233357](https://user-images.githubusercontent.com/30827440/39371279-c8e02c0a-4a73-11e8-858c-c215f5d95f5f.png)


it goes to this status in 5 hours

**Comment by pocesar** [comment]
may be related #748 (a new-old performance issue even on fast SSDs)

**Comment by ariesunny** [comment]
Restart this wallet， log shows as blow:

[2018-04-28 10:20:08.253867]: Bootstrap stopped because there are no peers
[2018-04-28 10:20:08.253867]: Bootstrap stopped because there are no peers
[2018-04-28 10:20:08.253867]: Exiting bootstrap attempt
[2018-04-28 10:20:08.621077]: Found a representative at [::ffff:45.32.246.108]:7075
[2018-04-28 10:20:08.861626]: Found a representative at [::ffff:144.217.167.119]:7075
[2018-04-28 10:20:09.377283]: Found a representative at [::ffff:192.82.109.90]:7075
[2018-04-28 10:20:09.392909]: Found a representative at [::ffff:188.226.156.108]:7075
[2018-04-28 10:20:09.561743]: Found a representative at [::ffff:163.172.179.193]:7075
[2018-04-28 10:20:09.762240]: Found a representative at [::ffff:178.22.66.84]:7075
[2018-04-28 10:20:09.762240]: Found a representative at [::ffff:178.22.66.84]:7075
[2018-04-28 10:20:10.147003]: 79 blocks in processing queue
[2018-04-28 10:20:11.994215]: Found a representative at [::ffff:104.225.223.205]:7075
[2018-04-28 10:20:12.031763]: Found a representative at [::ffff:18.220.173.108]:7075
[2018-04-28 10:20:12.063018]: Found a representative at [::ffff:174.138.4.198]:7075
[2018-04-28 10:20:12.448147]: Found a representative at [::ffff:103.105.50.84]:7075
[2018-04-28 10:20:13.048426]: Found a representative at [::ffff:139.162.205.25]:7075
[2018-04-28 10:20:13.248955]: Starting bootstrap attempt
[2018-04-28 10:20:13.433448]: Found a representative at [::ffff:45.55.67.202]:7075
[2018-04-28 10:20:13.622991]: frontier_req failed, reattempting
[2018-04-28 10:20:13.764772]: frontier_req failed, reattempting
[2018-04-28 10:20:13.888276]: frontier_req failed, reattempting
[2018-04-28 10:20:13.937301]: frontier_req failed, reattempting
[2018-04-28 10:20:14.349861]: Received 1 frontiers from [::ffff:192.99.176.122]:7075
[2018-04-28 10:20:26.558057]: 220 blocks in processing queue
[2018-04-28 10:20:29.436879]: Received 68060 frontiers from [::ffff:192.99.176.122]:7075
[2018-04-28 10:20:43.345668]: 98 blocks in processing queue
[2018-04-28 10:20:44.430301]: Received 137513 frontiers from [::ffff:192.99.176.122]:7075
[2018-04-28 10:20:59.435003]: Received 206001 frontiers from [::ffff:192.99.176.122]:7075
[2018-04-28 10:21:05.741507]: 69 blocks in processing queue
[2018-04-28 10:21:14.510425]: Received 403857 frontiers from [::ffff:192.99.176.122]:7075
[2018-04-28 10:21:27.848267]: 111 blocks in processing queue
[2018-04-28 10:21:29.509053]: Received 490543 frontiers from [::ffff:192.99.176.122]:7075
[2018-04-28 10:21:30.128744]: Completed frontier request, 477695 out of sync accounts according to [::ffff:192.99.176.122]:7075
[2018-04-28 10:21:44.667864]: 477685 accounts in pull queue
[2018-04-28 10:21:59.962189]: 477655 accounts in pull queue
[2018-04-28 10:22:12.217589]: 69 blocks in processing queue
[2018-04-28 10:22:14.968015]: 477595 accounts in pull queue
[2018-04-28 10:22:27.231709]: 5108 blocks in processing queue
[2018-04-28 10:22:30.134750]: 477520 accounts in pull queue
[2018-04-28 10:22:42.245241]: 5148 blocks in processing queue
[2018-04-28 10:22:45.516686]: 477429 accounts in pull queue
[2018-04-28 10:22:57.251935]: 6588 blocks in processing queue
[2018-04-28 10:23:00.553700]: 477290 accounts in pull queue
[2018-04-28 10:23:12.256714]: 7857 blocks in processing queue
[2018-04-28 10:23:15.656971]: 477108 accounts in pull queue
[2018-04-28 10:23:27.253366]: 7957 blocks in processing queue
[2018-04-28 10:23:30.868428]: 476917 accounts in pull queue
[2018-04-28 10:23:42.258167]: 7393 blocks in processing queue
[2018-04-28 10:23:45.927633]: 476737 accounts in pull queue
[2018-04-28 10:23:57.279918]: 8048 blocks in processing queue
[2018-04-28 10:24:01.080778]: 476509 accounts in pull queue
[2018-04-28 10:24:12.285723]: 8193 blocks in processing queue
[2018-04-28 10:24:16.102214]: 476290 accounts in pull queue
[2018-04-28 10:24:27.292002]: 13818 blocks in processing queue
[2018-04-28 10:24:31.313583]: 476047 accounts in pull queue
[2018-04-28 10:24:42.298174]: 14025 blocks in processing queue
[2018-04-28 10:24:46.538881]: 475859 accounts in pull queue
[2018-04-28 10:24:57.327509]: 14639 blocks in processing queue
[2018-04-28 10:25:01.551934]: 475614 accounts in pull queue
[2018-04-28 10:25:12.324641]: 14054 blocks in processing queue
[2018-04-28 10:25:16.680498]: 475402 accounts in pull queue
[2018-04-28 10:25:27.343822]: 14649 blocks in processing queue
[2018-04-28 10:25:31.710346]: 475132 accounts in pull queue
[2018-04-28 10:25:42.338231]: 14174 blocks in processing queue
[2018-04-28 10:25:46.807361]: 474894 accounts in pull queue
[2018-04-28 10:25:57.478775]: 14004 blocks in processing queue
[2018-04-28 10:26:01.866209]: 474663 accounts in pull queue
[2018-04-28 10:26:08.965400]: Found a representative at [::ffff:139.59.255.136]:7075
[2018-04-28 10:26:08.991155]: Found a representative at [::ffff:66.175.210.21]:7075
[2018-04-28 10:26:12.485666]: 14135 blocks in processing queue
[2018-04-28 10:26:16.908793]: 474361 accounts in pull queue
[2018-04-28 10:26:27.494775]: 14357 blocks in processing queue
[2018-04-28 10:26:31.933815]: 474026 accounts in pull queue
[2018-04-28 10:26:42.496733]: 16749 blocks in processing queue
[2018-04-28 10:26:46.928872]: 473658 accounts in pull queue
[2018-04-28 10:26:57.511963]: 16615 blocks in processing queue
[2018-04-28 10:27:01.967526]: 473335 accounts in pull queue
[2018-04-28 10:27:12.520941]: 49683 blocks in processing queue
[2018-04-28 10:27:16.991499]: 472987 accounts in pull queue
[2018-04-28 10:27:28.039671]: 73928 blocks in processing queue
[2018-04-28 10:27:32.020743]: 472689 accounts in pull queue
[2018-04-28 10:27:37.167041]: Found a representative at [::ffff:139.59.31.249]:7075
[2018-04-28 10:27:43.044245]: 77615 blocks in processing queue
[2018-04-28 10:27:47.035703]: 472285 accounts in pull queue
[2018-04-28 10:27:58.048972]: 79362 blocks in processing queue
[2018-04-28 10:28:02.065214]: 471925 accounts in pull queue
[2018-04-28 10:28:13.053179]: 80156 blocks in processing queue
[2018-04-28 10:28:17.073407]: 471461 accounts in pull queue
[2018-04-28 10:28:28.057394]: 83515 blocks in processing queue
[2018-04-28 10:28:32.134269]: 470970 accounts in pull queue
[2018-04-28 10:28:43.060627]: 85971 blocks in processing queue
[2018-04-28 10:28:47.144483]: 470401 accounts in pull queue
[2018-04-28 10:28:58.068340]: 87585 blocks in processing queue
[2018-04-28 10:29:02.152856]: 469857 accounts in pull queue
[2018-04-28 10:29:13.073509]: 87760 blocks in processing queue
[2018-04-28 10:29:17.175456]: 469475 accounts in pull queue
[2018-04-28 10:29:28.081406]: 90441 blocks in processing queue
[2018-04-28 10:29:32.212860]: 469022 accounts in pull queue
[2018-04-28 10:29:43.084819]: 91889 blocks in processing queue
[2018-04-28 10:29:47.285145]: 468520 accounts in pull queue
[2018-04-28 10:29:58.094284]: 92990 blocks in processing queue
[2018-04-28 10:30:02.352144]: 467969 accounts in pull queue
[2018-04-28 10:30:13.114804]: 103340 blocks in processing queue
[2018-04-28 10:30:17.385116]: 467457 accounts in pull queue
[2018-04-28 10:30:28.113799]: 103770 blocks in processing queue
[2018-04-28 10:30:32.446612]: 466938 accounts in pull queue
[2018-04-28 10:30:43.126748]: 106330 blocks in processing queue
[2018-04-28 10:30:47.453987]: 466325 accounts in pull queue
[2018-04-28 10:30:58.200452]: 107724 blocks in processing queue
[2018-04-28 10:31:02.486837]: 465698 accounts in pull queue
[2018-04-28 10:31:13.211851]: 107363 blocks in processing queue
[2018-04-28 10:31:17.494909]: 465042 accounts in pull queue
[2018-04-28 10:31:28.218379]: 108692 blocks in processing queue
[2018-04-28 10:31:32.505300]: 464397 accounts in pull queue
[2018-04-28 10:31:43.226047]: 113046 blocks in processing queue
[2018-04-28 10:31:47.542862]: 463622 accounts in pull queue
[2018-04-28 10:31:58.980815]: 119573 blocks in processing queue
[2018-04-28 10:32:02.551816]: 462889 accounts in pull queue
[2018-04-28 10:32:13.985408]: 137752 blocks in processing queue
[2018-04-28 10:32:16.669775]: Found a representative at [::ffff:35.224.233.216]:7075
[2018-04-28 10:32:17.606414]: 462236 accounts in pull queue
[2018-04-28 10:32:28.988753]: 142997 blocks in processing queue
[2018-04-28 10:32:32.689476]: 461711 accounts in pull queue
[2018-04-28 10:32:43.992967]: 146721 blocks in processing queue
[2018-04-28 10:32:47.695233]: 461161 accounts in pull queue
[2018-04-28 10:32:59.003553]: 147385 blocks in processing queue
[2018-04-28 10:33:02.704429]: 460537 accounts in pull queue
[2018-04-28 10:33:11.455521]: Found a representative at [::ffff:35.231.7.221]:7075
[2018-04-28 10:33:14.004514]: 148640 blocks in processing queue
[2018-04-28 10:33:17.707216]: 459854 accounts in pull queue
[2018-04-28 10:33:29.030045]: 148017 blocks in processing queue
[2018-04-28 10:33:32.757726]: 459189 accounts in pull queue
[2018-04-28 10:33:44.021399]: 144440 blocks in processing queue
[2018-04-28 10:33:47.807085]: 458546 accounts in pull queue
[2018-04-28 10:33:59.038736]: 142462 blocks in processing queue
[2018-04-28 10:34:02.855997]: 457856 accounts in pull queue
[2018-04-28 10:34:14.046099]: 135499 blocks in processing queue
[2018-04-28 10:34:17.866795]: 457185 accounts in pull queue
[2018-04-28 10:34:29.055340]: 134908 blocks in processing queue
[2018-04-28 10:34:32.979063]: 456461 accounts in pull queue
[2018-04-28 10:34:44.431731]: 133355 blocks in processing queue
[2018-04-28 10:34:48.003769]: 455872 accounts in pull queue
[2018-04-28 10:34:59.440532]: 201487 blocks in processing queue
[2018-04-28 10:35:03.031581]: 455121 accounts in pull queue
[2018-04-28 10:35:14.447255]: 294299 blocks in processing queue
[2018-04-28 10:35:18.049076]: 454403 accounts in pull queue
[2018-04-28 10:35:29.443191]: 430225 blocks in processing queue
[2018-04-28 10:35:33.099616]: 453852 accounts in pull queue
[2018-04-28 10:35:44.465160]: 507042 blocks in processing queue
[2018-04-28 10:35:48.107661]: 453190 accounts in pull queue
[2018-04-28 10:35:59.458582]: 583349 blocks in processing queue
[2018-04-28 10:36:03.145738]: 452534 accounts in pull queue
[2018-04-28 10:36:14.469702]: 714721 blocks in processing queue
[2018-04-28 10:36:18.356967]: 451838 accounts in pull queue
[2018-04-28 10:36:33.360661]: 451151 accounts in pull queue
[2018-04-28 10:36:38.064257]: 938607 blocks in processing queue
[2018-04-28 10:36:48.438107]: 450668 accounts in pull queue
[2018-04-28 10:36:53.290816]: 950720 blocks in processing queue
[2018-04-28 10:37:03.919167]: 450090 accounts in pull queue
[2018-04-28 10:37:08.285047]: 947439 blocks in processing queue
[2018-04-28 10:37:28.953492]: 449893 accounts in pull queue
[2018-04-28 10:37:31.524819]: 947825 blocks in processing queue
[2018-04-28 10:37:49.226965]: 449882 accounts in pull queue
[2018-04-28 10:37:51.983578]: 948021 blocks in processing queue
[2018-04-28 10:38:06.763448]: 449861 accounts in pull queue
[2018-04-28 10:38:09.312525]: 947776 blocks in processing queue
[2018-04-28 10:38:25.016892]: 449851 accounts in pull queue
[2018-04-28 10:38:26.750157]: 947369 blocks in processing queue
[2018-04-28 10:38:40.078217]: 449825 accounts in pull queue
[2018-04-28 10:38:41.755176]: 946612 blocks in processing queue
[2018-04-28 10:39:09.890345]: 449801 accounts in pull queue
[2018-04-28 10:39:14.907814]: 948594 blocks in processing queue

**Comment by ariesunny** [comment]
@PlasmaPower 
Please have a look of the logs:

It seems like most of the blocks pulled from bootstrapping will be checked as "gap_previous" or “gap_source”, then these blocks will be kept in the unchecked_cache until the missing "previous" or "source" was received and checked successfully, and then they can been re-processed. But how can the network make sure this node will definitely re-received the missing "previous" or "source" blocks for no nodes are rebroadcasting them?

```
[2018-04-28 16:28:03.802185]: Pulled block BE7D882DC93B4A3961FE380B57FD81AB39CA96A36F8211DD6C78D2CC4CD8BCBC {
    "type": "send",
    "previous": "A80AD06C99F30FCBCB6FF5F8DA7F95355E06B91EF4041F2E951B657F2A952B31",
    "destination": "xrb_117ikxnfcpoz7oz56sxdw5yhwtt86rrwxsg5io6c6pbqa4fsr1cnyot4ikqu",
    "balance": "00000000000000000000000000000000",
    "work": "e26016993ec76099",
    "signature": "E676AEC583D316960BB4A5F5312907700CBAE149B136C0D7350E9C962D8133CB0DEA26D73D75A8E83B2B98253DB7D25F7AB19881A6A9AB0FC3160B6E5C023602"
}

[2018-04-28 16:28:03.802185]: Gap previous for: BE7D882DC93B4A3961FE380B57FD81AB39CA96A36F8211DD6C78D2CC4CD8BCBC
[2018-04-28 16:28:03.802185]: Received confirm_ack message from [::ffff:138.68.165.105]:7075 for 73270F726D2A0E344FD337D6072CB3CC225FA02154135A570794063BA45E15E3 sequence 358679737
[2018-04-28 16:28:03.802185]: Pulled block A80AD06C99F30FCBCB6FF5F8DA7F95355E06B91EF4041F2E951B657F2A952B31 {
    "type": "receive",
    "previous": "C407F21FB9FD602FE78D09ECBF875442AF08E2B7E74753513A2A2E16656A5F4B",
    "source": "19A3618501C88BE5D140206E88A9F8C1F6DA95BF0C04ADE3E0FB0A4C2D0D80D4",
    "work": "9136656b7fef9c75",
    "signature": "1CF84FD1C3A37C3CC06E4870C9B639BF83785E2632BABEF23679E4BC95E66A5832E9FBB990B903FEA25628F5F8820756ABB1973FCFB83E3CF7526C9D8FFC8506"
}

[2018-04-28 16:28:03.802185]: Gap previous for: A80AD06C99F30FCBCB6FF5F8DA7F95355E06B91EF4041F2E951B657F2A952B31
[2018-04-28 16:28:03.802185]: Pulled block C407F21FB9FD602FE78D09ECBF875442AF08E2B7E74753513A2A2E16656A5F4B {
    "type": "send",
    "previous": "38A3A4B9C2488D0BE351C89A4B69C8DFF7D6116153FEBE5387D5477644D8B0AC",
    "destination": "xrb_117ikxnfcpoz7oz56sxdw5yhwtt86rrwxsg5io6c6pbqa4fsr1cnyot4ikqu",
    "balance": "00000000000000000000000000000000",
    "work": "68580f20959092bb",
    "signature": "3655744BC174D60AD9BF4566AEBF336A4DC9F88AA123F6DC87AABE2853F349D089E5EFFD90FFC3CE1FED6058D559AF3D684E954F93FB0D3AFEE4881133709603"
}

[2018-04-28 16:28:03.802185]: Gap previous for: 73270F726D2A0E344FD337D6072CB3CC225FA02154135A570794063BA45E15E3
[2018-04-28 16:28:03.802185]: Vote from: xrb_1hza3f7wiiqa7ig3jczyxj5yo86yegcmqk3criaz838j91sxcckpfhbhhra1 sequence: 358679737 block: 73270F726D2A0E344FD337D6072CB3CC225FA02154135A570794063BA45E15E3 status: Vote
[2018-04-28 16:28:03.802185]: Gap previous for: C407F21FB9FD602FE78D09ECBF875442AF08E2B7E74753513A2A2E16656A5F4B
[2018-04-28 16:28:03.803188]: Received confirm_ack message from [::ffff:188.166.98.45]:7075 for 73270F726D2A0E344FD337D6072CB3CC225FA02154135A570794063BA45E15E3 sequence 358679737
[2018-04-28 16:28:03.803188]: Gap previous for: 73270F726D2A0E344FD337D6072CB3CC225FA02154135A570794063BA45E15E3
[2018-04-28 16:28:03.803188]: Vote from: xrb_1hza3f7wiiqa7ig3jczyxj5yo86yegcmqk3criaz838j91sxcckpfhbhhra1 sequence: 358679737 block: 73270F726D2A0E344FD337D6072CB3CC225FA02154135A570794063BA45E15E3 status: Vote
[2018-04-28 16:28:03.804194]: Received confirm_ack message from [::ffff:206.189.18.189]:7075 for 9F89D1B27AFCAE09AE9E8D14600F5E088813EE52FC2D62166FE479AD5999B5ED sequence 358679757
[2018-04-28 16:28:03.804194]: Gap previous for: 9F89D1B27AFCAE09AE9E8D14600F5E088813EE52FC2D62166FE479AD5999B5ED
[2018-04-28 16:28:03.804194]: Vote from: xrb_1hza3f7wiiqa7ig3jczyxj5yo86yegcmqk3criaz838j91sxcckpfhbhhra1 sequence: 358679757 block: 9F89D1B27AFCAE09AE9E8D14600F5E088813EE52FC2D62166FE479AD5999B5ED status: Vote
[2018-04-28 16:28:03.805188]: Received confirm_ack message from [::ffff:78.46.212.33]:7075 for A1A702E886FBABB0B6C0E9F799ED93A46B60B1138E7CD164C9FA85C7F0FEA60A sequence 520128094
[2018-04-28 16:28:03.805188]: Gap previous for: A1A702E886FBABB0B6C0E9F799ED93A46B60B1138E7CD164C9FA85C7F0FEA60A
[2018-04-28 16:28:03.805188]: Vote from: xrb_1anrzcuwe64rwxzcco8dkhpyxpi8kd7zsjc1oeimpc3ppca4mrjtwnqposrs sequence: 520128094 block: A1A702E886FBABB0B6C0E9F799ED93A46B60B1138E7CD164C9FA85C7F0FEA60A status: Vote
[2018-04-28 16:28:03.806206]: Received confirm_ack message from [::ffff:104.131.91.161]:7075 for 73270F726D2A0E344FD337D6072CB3CC225FA02154135A570794063BA45E15E3 sequence 520128086
[2018-04-28 16:28:03.806206]: Gap previous for: 73270F726D2A0E344FD337D6072CB3CC225FA02154135A570794063BA45E15E3
[2018-04-28 16:28:03.806206]: Vote from: xrb_1anrzcuwe64rwxzcco8dkhpyxpi8kd7zsjc1oeimpc3ppca4mrjtwnqposrs sequence: 520128086 block: 73270F726D2A0E344FD337D6072CB3CC225FA02154135A570794063BA45E15E3 status: Vote
[2018-04-28 16:28:03.806206]: Received confirm_ack message from [::ffff:45.55.11.199]:7075 for 67416C0662F2E834B783EB3600AC74EF466F16DE4A0C883500412091A665C749 sequence 520128092
[2018-04-28 16:28:03.806206]: Gap previous for: 67416C0662F2E834B783EB3600AC74EF466F16DE4A0C883500412091A665C749
[2018-04-28 16:28:03.806206]: Vote from: xrb_1anrzcuwe64rwxzcco8dkhpyxpi8kd7zsjc1oeimpc3ppca4mrjtwnqposrs sequence: 520128092 block: 67416C0662F2E834B783EB3600AC74EF466F16DE4A0C883500412091A665C749 status: Vote
[2018-04-28 16:28:03.810285]: Received confirm_ack message from [::ffff:50.116.6.204]:7075 for 73270F726D2A0E344FD337D6072CB3CC225FA02154135A570794063BA45E15E3 sequence 358679737
[2018-04-28 16:28:03.811289]: Gap previous for: 73270F726D2A0E344FD337D6072CB3CC225FA02154135A570794063BA45E15E3
[2018-04-28 16:28:03.811289]: Vote from: xrb_1hza3f7wiiqa7ig3jczyxj5yo86yegcmqk3criaz838j91sxcckpfhbhhra1 sequence: 358679737 block: 73270F726D2A0E344FD337D6072CB3CC225FA02154135A570794063BA45E15E3 status: Vote
[2018-04-28 16:28:03.811289]: Received confirm_ack message from [::ffff:188.68.32.148]:7075 for 003DF5A50139FBD1F896705E35AC575E04F248B498801CC5961FAC2CBE2E3326 sequence 520128114
[2018-04-28 16:28:03.811289]: Gap previous for: 003DF5A50139FBD1F896705E35AC575E04F248B498801CC5961FAC2CBE2E3326
```

**Comment by PlasmaPower** [comment]
> how can the network make sure this node will definitely re-received the missing "previous" or "source" blocks for no nodes are rebroadcasting them?

The node already has the missing blocks in unchecked but it hasn't processed them yet.

**Comment by ariesunny** [comment]
@PlasmaPower 
Yes, the node has them in the unchecked list，but it needs a trigger to re-process unchecked and move them to SEND/RECEIVE/OPEN/CHANGE list as checked blocks，Right? What is the mechanics of it? 
Thank you!

**Comment by PlasmaPower** [comment]
When a block is processed, any blocks that depend on it (and were previously missing that block) are processed next.

**Comment by ariesunny** [comment]
@PlasmaPower Back to the former question: "how can the network make sure this node will definitely re-received the missing "previous" or "source" blocks for no nodes are rebroadcasting them?"

**Comment by PlasmaPower** [comment]
It simply pulls all blocks of other nodes. If those nodes have a block, they will have its dependency.

**Comment by ariesunny** [comment]
@PlasmaPower Thank you, Yes, the mechanics works. The current state of my node is that all the blocks in my node is 832555 + 7621827 = 8,454,382 ,while the blocks of whole network is 8,791,060, which means there are 300000 blocks not pulled from other node, and it seems the following 300000 blocks download very slowly.   What's reason may be ? Any idea how to accelerate it ?

**Comment by rkeene** [comment]
With the changes added recently in V17.0 with regards to Lazy Bootstrapping and improvements with fork resolution, I believe this issue is resolved in V17.0.  Please re-open if this issue persists with V17 !

Thanks !
